fitbenchmarking/controllers/gsl_controller.py

```diff
@@ -207,18 +206,24 @@ def fit(self):
                 status = multiminimize.test_gradient(gradient,
                                                      self._gradient_tol)
             if status == errno.GSL_SUCCESS:
-                self.success = True
+                self._status = 0
                 break
             elif status != errno.GSL_CONTINUE:
-                raise RuntimeError("GSL couldn't find a solution")
+                self._status = 2
         else:
-            raise RuntimeError("Maximum number of iterations exceeded in GSL")
+            self._status = 1
 
     def cleanup(self):
         """
         Convert the result to a numpy array and populate the variables results
         will be read from
         """
-        if self.success:
-            self.final_params = self._solver.getx()
-            self.results = self.problem.eval_f(params=self.final_params)
+        if self._status == 0:
+            self.flag = 0
+        elif self._status == 1:
+            self.flag = 1
+        else:
+            self.flag = 2
```

Comment:
remove `self._status` from this (as in comments above)

Reply:
Done